server/src/main/java/org/elasticsearch/index/translog/TranslogHeader.java

```diff
@@ -105,6 +106,21 @@ private static int headerSizeInBytes(int version, int uuidLength) {
         return size;
     }
 
+
+    static int readHeaderVersion(final Path path, final FileChannel channel, final StreamInput in) throws IOException {
```

Comment:
can you forward-port this method extraction so that we have code more aligned on master/7.x?

Reply:
Sure, will do.